lib/cosmos/tools/tlm_grapher/tabbed_plots_tool/tabbed_plots_realtime_thread.rb

```diff
@@ -65,6 +65,12 @@ def kill
 
     def graceful_kill
       # Just to remove warning
+      if defined? Qt and Thread.current == Thread.main
+        5.times do
```

Comment:
Perhaps a note to say what you're trying to accomplish here. I think you're allowing the QT application to process all it's GUI events before dying. Why not just kill it hard?